pkg/api/kptfile/v1/types.go

```diff
@@ -20,17 +20,35 @@ package v1
 import (
```

Comment:
@yuwenma Why does the kpt-functions-sdk have a copy of the Kptfile type? It seems like the kpt repo would be the natural place to put it.

Reply:
I don't know the exact legacy reasons, but I think it makes sense for 2 reasons
1. Users who need to use the Kptfile type (i.e. to write their own kptfile relevant functions), they can import from go-sdk and do not need to import the entire kpt pkg.
2.  kpt, as a CLI, is not designed for importing purpose so it should not worry too much about supporting its dependants. 


> It seems like the kpt repo would be the natural place have it and the sdk would import it.

I see kpt/porch has imported go/fn in several places. e.g. [porch function wrapper main.go](https://github.com/GoogleContainerTools/kpt/blob/main/porch/func/wrapper-server/main.go#L26),  [porch builtin function runtime](https://github.com/GoogleContainerTools/kpt/blob/cdb66f373d6c485ecc7f36de414e58296e065c89/porch/pkg/engine/builtinruntime.go#L24), which is aligned with the library dependencies I described above. And because of that, I don't think sdk can then import kpt/porch.